src/routes/__tests__/DisciplineSelectionScreen.spec.tsx

```diff
@@ -0,0 +1,69 @@
+import { RouteProp } from '@react-navigation/native'
+import { fireEvent, waitFor } from '@testing-library/react-native'
+import { mocked } from 'jest-mock'
+import React from 'react'
+
+import { useLoadDisciplines } from '../../hooks/useLoadDisciplines'
+import useReadSelectedProfessions from '../../hooks/useReadSelectedProfessions'
+import { RoutesParams } from '../../navigation/NavigationTypes'
+import createNavigationMock from '../../testing/createNavigationPropMock'
+import { getReturnOf } from '../../testing/helper'
+import { mockDisciplines } from '../../testing/mockDiscipline'
+import render from '../../testing/render'
+import DisciplineSelectionScreen from '../DisciplineSelectionScreen'
+
+jest.mock('@react-navigation/native')
+jest.mock('../../hooks/useLoadDisciplines')
+jest.mock('../../hooks/useReadSelectedProfessions')
+
+describe('DisciplineSelectionScreen', () => {
+  const navigation = createNavigationMock<'DisciplineSelection'>()
+  const getRoute = (): RouteProp<RoutesParams, 'DisciplineSelection'> => ({
+    key: 'key-1',
+    name: 'DisciplineSelection',
+    params: {
+      discipline: mockDisciplines()[1],
+    },
+  })
+
+  it('should display the correct title', () => {
+    mocked(useLoadDisciplines).mockReturnValueOnce(getReturnOf(mockDisciplines()))
+    mocked(useReadSelectedProfessions).mockReturnValueOnce(getReturnOf([mockDisciplines()[0].id]))
+
+    const { getByText } = render(<DisciplineSelectionScreen route={getRoute()} navigation={navigation} />)
+    const title = getByText(mockDisciplines()[0].title)
+    expect(title).toBeDefined()
+  })
+
+  it('should display all disciplines', async () => {
+    mocked(useLoadDisciplines).mockReturnValueOnce(getReturnOf(mockDisciplines()))
+    mocked(useReadSelectedProfessions).mockReturnValueOnce(getReturnOf([mockDisciplines()[0].id]))
+
+    const { getByText, findByText } = render(<DisciplineSelectionScreen route={getRoute()} navigation={navigation} />)
+
+    const firstDiscipline = await findByText(mockDisciplines()[0].title)
+    const secondDiscipline = await findByText(mockDisciplines()[0].title)
+    const thirdDiscipline = getByText(mockDisciplines()[0].title)
```

Comment:
```suggestion
    const firstDiscipline = getByText(mockDisciplines()[0].title)
    const secondDiscipline = getByText(mockDisciplines()[1].title)
    const thirdDiscipline = getByText(mockDisciplines()[2].title)
```

Reply:
I don't know why, but multiple elements are found for the second  discipline

Reply:
Removing what's not needed, I guess :)

Reply:
I guess that's because The Discipline selectionScreen shows the child disciplines of a discipline in a list, in the getRoute() mock the second mock discipline is set as the parent discipline. Then you also say that useLoadDiscipline returns all the mock discipline. Therefore the second discipline appears twice